blocks/autocompleters/index.js

```diff
@@ -51,55 +63,54 @@ import BlockIcon from '../block-icon';
  * @typedef {Object} Completer
  * @property {?String} className A class to apply to the popup menu.
  * @property {String} triggerPrefix the prefix that will display the menu.
- * @property {FnGetOptions} getOptions get the block options in a resolved promise.
+ * @property {FnGetOptions} getOptions get the completer options in a resolved promise.
+ * @property {?FnGetOptionKeywords} getOptionKeywords get the keywords for a given option.
+ * @property {FnRenderOptionLabel} renderOptionLabel render the label for a given option.
  * @property {?FnAllowNode} allowNode filter the allowed text nodes in the autocomplete.
  * @property {?FnAllowContext} allowContext filter the context under which the autocomplete activates.
- * @property {FnOnSelect} onSelect
+ * @property {FnGetOptionResult} getOptionResult
  */
 
 /**
  * Returns an "completer" definition for selecting from available blocks to replace the current one.
  * The definition can be understood by the Autocomplete component.
  *
- * @param {Function} onReplace  Callback to replace the current block.
- *
  * @return {Completer} Completer object used by the Autocomplete component.
  */
-export function blockAutocompleter( { onReplace } ) {
-	// Prioritize common category in block type options
-	const options = sortBy(
-		getBlockTypes(),
-		( { category } ) => 'common' !== category
-	).map( ( blockType ) => {
-		const { name, title, icon, keywords = [] } = blockType;
-		return {
-			value: name,
-			label: [
-				<BlockIcon key="icon" icon={ icon } />,
-				title,
-			],
-			keywords: [ ...keywords, title ],
-		};
-	} );
-
-	const getOptions = () => Promise.resolve( options );
+export function blockAutocompleter() {
+	let promisedOptions;
+
+	const getOptions = () => promisedOptions || ( promisedOptions = Promise.resolve(
```

Comment:
If you want to make sure this is called only once you can use `lodash.once`
```js
const getOptions = once( () => Promise.resolve... );
```

Reply:
Thanks for the tip! That's cleaner. 
I'll make the update.